### ISSUE TITLE
Watcher crashes but server keeps running

### ISSUE THREAD

**Issue by brunvez** [phoenixframework/phoenix#3996]
Related to https://github.com/phoenixframework/phoenix/issues/1870, when developing with autosave mode many times incomplete code gets saved and when Webpack tries to transpile it it crashes. When this happens the Phoenix server keeps running, which is bothersome because you have to manually kill it and restart the whole thing.

This might be more related to Webpack than Phoenix but a solution that could improve dev experience could be to link the process status to the watcher, wait a few seconds, and restart it. I'd love to give it a try if pointed in the right direction since I've only found ways to do this with external dependencies, which even in dev I think might be a bit of an overkill.

### Environment

* Elixir version (elixir -v): Elixir 1.10.1
* Phoenix version (mix deps): 1.5.3
* NodeJS version (node -v): v12.16.1
* NPM version (npm -v): 6.13.4
* Operating system: macOS Mojave 10.14.6 

### Expected behavior

When the Webpack server crashes on dev it should either crash the server or (better yet) be restarted.

### Actual behavior

The Webpack server crashes but the Phoenix server keeps serving requests, which causes the assets not to be served.

![image](https://user-images.githubusercontent.com/17035407/93691432-7c551300-fabb-11ea-9848-e4b409fca169.png)
![image](https://user-images.githubusercontent.com/17035407/93691436-86771180-fabb-11ea-9a44-8aa8ed73461b.png)

**Comment by brunvez** [comment]
Did some more homework, turns out the Webpack task does not exit upon error so there's no way to get the error code. I tried using ports to monitor it but with no luck. Seems like it's really more of a Webpack/Babel issue, especially since if there's an error while first building the dev server will be resilient and actually restart itself, while if the error happens after Webpack will not try to re-build. Or at least that's my case.

I've also checked that other node scripts do get restarted with this simple snippet which gets retried every 3 seconds

```javascript
setTimeout(function () {
  throw new Error()
}, 3000)
```

Anyway, just leaving what I found here for documentation purposes. Feel free to close it if there's no other solution that could help prevent this, it's still kind of annoying to restart the server every couple of minutes.
